### PR TITLE
WideOrbit adapter

### PR DESCRIPTION
**Pull request by bjorn-wo**
Hi!

We are currently running live tests with our fork on a select number of customers but we would need to make it into the official branch.

How do we update your documentation with our adapter parameters?

Best regards
Björn Andersson
WideOrbit Programmatic Digital 
http://www.wideorbit.com/wideorbit-products/woprogrammatic-digital/

### CODE REVIEW

File: src/adaptermanager.js [hunk 1]
```diff
@@ -70,8 +70,8 @@ exports.aliasBidAdapter = function (bidderCode, alias) {
           utils.logError(bidderCode + ' bidder does not currently support aliasing.', 'adaptermanager.aliasBidAdapter');
         } else {
           newAdapter = bidAdaptor.createNew();
-          newAdapter.setBidderCode(alias);
-          this.registerBidAdapter(newAdapter, alias);
+					newAdapter.setBidderCode(alias);
```
Comment (mkendall07) [inline src/adaptermanager.js]: please fix formatting

File: src/adapters/wideorbit.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    bidmanager = require('../bidmanager.js'),
+    utils = require('../utils.js'),
+    adloader = require('../adloader');
+
+var WideOrbitAdapter = function WideOrbitAdapter() {
+  var pageImpression = 'JSAdservingMP.ashx?pc={pc}&pbId={pbId}&clk=&exm=&jsv=1.0&tsv=1.0&cts={cts}&arp=0&fl=0&vitp=&vit=&jscb=window.parent.pbjs.handleWideOrbitCallback&url=&fp=&oid=&exr=&mraid=&apid=&apbndl=&mpp=0&uid=&cb={cb}&hb=1',
+      pageRepeatCommonParam = '&gid{o}={gid}&pp{o}=&clk{o}=&rpos{o}={rpos}&ecpm{o}={ecpm}&ntv{o}=&ntl{o}=&adsid{o}=',
+      pageRepeatParamId = '&pId{o}={pId}&rank{o}={rank}',
+      pageRepeatParamNamed = '&wsName{o}={wsName}&wName{o}={wName}&rank{o}={rank}&bfDim{o}={width}x{height}&subp{o}={subp}',
+      base = (window.location.protocol) + '//p{pbId}.atemda.com/',
```
Comment (mkendall07) [inline src/adapters/wideorbit.js]: please hardcode to `HTTPS`

File: src/adapters/wideorbit.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
+var bidfactory = require('../bidfactory.js'),
+    bidmanager = require('../bidmanager.js'),
+    utils = require('../utils.js'),
+    adloader = require('../adloader');
+
+var WideOrbitAdapter = function WideOrbitAdapter() {
+  var pageImpression = 'JSAdservingMP.ashx?pc={pc}&pbId={pbId}&clk=&exm=&jsv=1.0&tsv=1.0&cts={cts}&arp=0&fl=0&vitp=&vit=&jscb=window.parent.pbjs.handleWideOrbitCallback&url=&fp=&oid=&exr=&mraid=&apid=&apbndl=&mpp=0&uid=&cb={cb}&hb=1',
```
Comment (mkendall07) [inline src/adapters/wideorbit.js]: Is there a reason to use `window.parent` here instead of just `window`?

**Comment by mkendall07** [comment]
@bjorn-wo 
Thank you for the PR. Can you please update `/integrationExamples/gpt/pbjs_example_gpt.html` with your adapter example (https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/pbjs_example_gpt.html).
Also please include test params to validate a bid response from the server. You can post them in a comment or email us directly (email in github profile). 

You can make a PR here to update the documentation:
https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders.md

File: src/adapters/wideorbit.js (hunk 2 above)
Comment (bjorn-wo) [inline src/adapters/wideorbit.js]: https is an expensive operation, also cost-wise for us. Is there some compatibility issue we overlook or similar? Please explain why we need to change this..

File: src/adapters/wideorbit.js (hunk 3 above)
Comment (bjorn-wo) [inline src/adapters/wideorbit.js]: This was a mistake by us, thank you for noticing.

File: src/adapters/wideorbit.js (hunk 2 above)
Comment (mkendall07) [inline src/adapters/wideorbit.js]: Generally we like to use `HTTPS` in all cases as it is the recommended way to load resources. Also it's simpler to just use one protocol. We can make an exception if this is really a concern for you,however, it has not been an issue for any other adapters.

File: integrationExamples/gpt/pbjs_example_gpt.html [hunk 4]
```diff
@@ -240,6 +240,13 @@
                             publisher_id: 5000563,  // REQUIRED int or str publisher ID. To get one, register at https://control.adequant.com
                             bidfloor: 0.01,         // OPTIONAL float bid floor in $ CPM
                         }
+                    },
+                    {
+                        bidder: 'wideorbit',
+                        params: {
+                            pbId: 123, // Publisher Id,
+                            pId: 123456 // Placement Id
```
Comment (mkendall07) [inline integrationExamples/gpt/pbjs_example_gpt.html]: Are both params required? Please mark in the comments

File: src/adapters/wideorbit.js (hunk 2 above)
Comment (bjorn-wo) [inline src/adapters/wideorbit.js]: Would be glad if you could. Actually, it seems like the appnexus, triplelift,springserve, rubicon, pulsepoint, indexExchange and nginad  adapters are also using similar checks.

File: integrationExamples/gpt/pbjs_example_gpt.html (hunk 4 above)
Comment (bjorn-wo) [inline integrationExamples/gpt/pbjs_example_gpt.html]: Have marked them as required. The usage of the parameters will be clear from the documentation. The user will also download those definitions from our UI or API, they will probably never create them manually.

File: package.json [hunk 5]
```diff
@@ -91,13 +92,16 @@
...
+    "querystringify": "0.0.3",
     "raw-loader": "^0.5.1",
     "redis": "^0.12.1",
     "requirejs": "^2.1.20",
     "run-sequence": "^1.1.4",
     "sinon": "^1.12.1",
     "string-replace-webpack-plugin": "0.0.3",
     "uglify-js": "^2.4.15",
+    "url-parse": "^1.0.5",
+    "webdriverio": "^3.2.5",
```
Comment (mkendall07) [inline package.json]: @bjorn-wo 

Can you remove this dev dependency as we aren't setup yet for automated testing.

**Comment by mkendall07** [comment]
@bjorn-wo 
It looks after our internal refactor some of your unit tests for the adapter are failing. Could you please update the PR with the working unit tests? Thanks!

File: src/adapters/wideorbit.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+    _processUserMatchings(response.UserMatchings);
+
+    utils._each(bids, function (bid) {
+      bidResponse = _getBidResponse(bid.placementCode, response.Placements);
+
+      if (bidResponse && bidResponse.Type === 'DirectHTML') {
+        bidObject = bidfactory.createBid(1);
+        bidObject.cpm = bidResponse.Bid;
+        bidObject.ad = _buildAdCode(bidResponse);
```
Comment (mkendall07) [inline src/adapters/wideorbit.js]: the `bid.ad` should contain the HTML/script markup of the ad. When I tested this, I got empty string for this param, which is invalid. It looks like you might need to get the `MaterialUrl` or `source` param here. If `source` is empty, it should not be a valid bid. Use `bidfactory.createBid(constants.STATUS.NO_BID)` instead.

Comment (bjorn-wo) [inline src/adapters/wideorbit.js]: Matt, I'm sorry for this. It's not a problem with the adapter, but the test data you got which returns the wrong data structure. We will fix this, I'll get back to you on email with proper parameters.

File: package.json (hunk 5 above)
Comment (bjorn-wo) [inline package.json]: Hi! wedriverio seems to be a leftover from your previous version, before refactoring. Perhaps a merge issue left it still there.

url-parse we need for our testing code.

**Comment by mkendall07** [comment]
Reviewed & LGTM.

**Pull request merged**